article/app/views/support/structuredData/BlogPosting.scala

```diff
@@ -0,0 +1,59 @@
+package views.support.structuredData
+
+import common.LinkTo
+import controllers.LiveBlogPage
+import model.liveblog.BodyBlock
+import org.joda.time.DateTime
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.RequestHeader
+import views.support.Format
+
+object BlogPosting {
+
+  def zulu(date: DateTime)(implicit request: RequestHeader): String = Format(date, "yyyy-MM-dd'T'HH:mm:ssZ")
+
+  def apply(blog: LiveBlogPage, block: BodyBlock)(implicit request: RequestHeader): JsValue = {
+
+    def blockDate(block: BodyBlock) = block.publishedDate match {
+      case Some(date) => zulu(date)
+      case None => zulu(blog.article.trail.webPublicationDate)
+    }
+
+    def blockBody(block: BodyBlock): String = {
+      if(block.bodyTextSummary.length < 1000){
+        block.bodyTextSummary
+      } else {
+        block.bodyTextSummary.substring(0, 1000)
+      }
+    }
+
+    def blockAuthor(blog: LiveBlogPage, block: BodyBlock): JsValue = {
+
+      val name = block.contributors.headOption match {
+        case Some(id) => blog.article.tags.tags.find(_.id == s"profile/$id").map{ contributorTag =>
+          contributorTag.name
+        }
+        case None => None
+      }
+
+      name match {
+        case Some(thing) => Json.obj("@type" -> "Person", "name" -> thing)
+        case None => Organisation()
+      }
+
+    }
+
+    Json.obj(
+      "@type" -> "BlogPosting",
+      "headline" -> block.title.getOrElse[String](blog.article.trail.headline),
+      "author" -> blockAuthor(blog, block),
+      "publisher" -> Organisation(),
```

Comment:
:man_facepalming: 

So just reading about it, I think you can maybe do this (tested it in the validator with their example and it seems to link it, (but there's so many errors in the original example...), like this:

```
"publisher": {
      	"@id": "http://awesometechblog.com/2015/03/09/acme_spring_forward/#publisher"
      },
```
and use that as the id of a different `json+ld` block:
```
 "@id": "http://awesometechblog.com/2015/03/09/acme_spring_forward/#publisher",
```

Full example - I only added publisher to the first block:

```json
<script type="application/ld+json">
{
    "@id": "http://awesometechblog.com/2015/03/09/acme_spring_forward/#publisher",
    "@context": "http://schema.org",
    "@type": "Organization",
    "name": "The Guardian"
    }
</script>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "LiveBlogPosting",
  "url": "http://awesometechblog.com/2015/03/09/acme_spring_forward",
  "about": {
    "@type": "Event",
    "startDate": "2015-03-09T13:00:00-07:00",
    "name": "Acme Spring Forward Event",
    "location": {
      "@type": "EventVenue",
      "name": "Jane Gardner",
      "address" :{
        "@type": "PostalAddress",
        "streetAddress": "701 Mission St",
        "addressLocality": "San Francisco",
        "addressRegion": "CA",
        "postalCode": "94103",
        "addressCountry": "US"
      }
    }
  },
  "coverageStartTime": "2015-03-09T11:30:00-07:00",
  "coverageEndTime": "2015-03-09T16:00:00-07:00",
  "headline": "Acme Spring Forward Event Live Blog",
  "description": "Welcome to Foo's live coverage of the Acme Spring Forward …",
  "liveBlogUpdate": [
    {
      "@type": "BlogPosting",
      "headline": "Coming this April, HBO NOW will be available exclusively in the U.S. on Acme TV and the App Store.",
      "url": "http://awesometechblog.com/2015/03/09/acme_spring_forward#post3",
      "datePublished": "2015-03-09T13:08:00-07:00",
      "publisher": {
      	"@id": "http://awesometechblog.com/2015/03/09/acme_spring_forward/#publisher"
      },
      "articleBody": {
        "@value": "It's $14.99 a month. <br> And for a limited time, only $9.99",
        "@type": "rdf:HTML"
      }
    },
    {
      "@type": "BlogPosting",
      "headline": "iPhone is growing at nearly twice the rate of the rest of the smartphone market.",
      "datePublished": "2015-03-09T13:13:00-07:00",
"image": "http://images.acme.com/live/2015-mar-event/images/573cb_xlarge_2x.jpg"
    },
    {
      "@type": "BlogPosting",
      "headline": "See the new flagship Acme Retail Store in West Lake, China.",
      "url": "http://awesometechblog.com/2015/03/09/acme_spring_forward#post1",
      "datePublished": "2015-03-09T13:17:00-07:00",
      "video": {
        "@type":"VideoObject",
"thumbnail": "http://images.acme.com/live/2015-mar-event/images/908d2e_large_2x.jpg"
      }
    }
  ]
}
</script>
```